spec/02-integration/03-db/07-declarative_spec.lua

```diff
@@ -0,0 +1,175 @@
+local Declarative = require "kong.db.declarative"
+local ssl_fixtures = require "spec.fixtures.ssl"
+local helpers = require "spec.helpers"
+local conf_loader = require "kong.conf_loader"
+
+for _, strategy in helpers.each_strategy() do
+  describe("declarative config #" .. strategy, function()
+    local db
+    lazy_setup(function()
+      local _
+      _, db = helpers.get_db_utils(strategy, {
+        "acls",
+        "plugins",
+        "routes",
+        "services",
+        "snis",
+        "certificates",
+        "consumers",
+      })
+      _G.kong.db = db
+      assert(helpers.start_kong({
+        database   = strategy,
+      }))
+    end)
+
+    lazy_teardown(function()
+      assert(helpers.stop_kong())
+    end)
+
+    before_each(function()
+      db.acls:truncate()
+      db.plugins:truncate()
+      db.routes:truncate()
+      db.services:truncate()
+      db.snis:truncate()
+      db.certificates:truncate()
+      db.consumers:truncate()
+    end)
```

Comment:
We should avoid this pattern of truncating on `before_each`. It's better to write tests that don't require truncating the database (i.e. make different tests make unrelated uuids/entities). If you _need_ to truncate a database (i.e. to verify a table remains empty) do so only for the needed tables in the specific tests.

With this `before_each` block it will incur in a significant time penalty for the tests: each time a new test is added to this file, even if it has nothing to do with some of these tables. And currently, with only one test, it is truncating everything twice, since `start_kong` is already doing it in `lazy_setup`.

(Truncating the necessary tables only at the beginning of a test file is fine to ensure a clean-slate, as `start_kong` above does already).